test/integration/model/bulk-create.test.js

```diff
@@ -706,6 +706,67 @@ describe(Support.getTestDialectTeaser('Model'), () => {
             expect(people[1].system).to.equal('system2');
             expect(people[1].name).to.equal('Bob');
           });
+
+          it('when the primary key column names and model field names are different and have composite unique index constraints', async function () {
+            const Person = this.sequelize.define(
+              'Person',
+              {
+                id: {
+                  type: DataTypes.INTEGER,
+                  allowNull: false,
+                  primaryKey: true,
+                  field: 'id'
+                },
+                systemId: {
+                  type: DataTypes.INTEGER,
+                  allowNull: false,
+                  field: 'system_id'
+                },
+                system: {
+                  type: DataTypes.STRING,
+                  allowNull: false,
+                  field: 'system'
+                },
+                name: {
+                  type: DataTypes.STRING,
+                  allowNull: false,
+                  field: 'name'
+                }
+              },
+              {
+                indexes: [
+                  {
+                    unique: true,
+                    fields: ['system_id', 'system']
+                  }
+                ]
+              }
```

Comment:
This fix would also work for when a unique index is defined like this:

```
const uniqueIndex = {
  name: 'user_friendship',
  msg: 'users are already friends',
};

export class Friendship extends Model {
  public senderId!: string;

  public receiverId!: string;

  public status!: FriendshipStatus;
}

```js
const uniqueIndex = {
  name: 'user_contest',
  msg: 'User is already part of contest.',
};

export class UserContestInfo extends Model {
  public contestId!: string;

  public prize?: number;

  public userId!: string;

  public rank?: number;
}

const UserContestInfoFactory = (connection: sequelize.Sequelize) => {
  const attributes: ModelAttributes = {
    userId: {
      type: DataTypes.UUID,
      unique: uniqueIndex,
    },
    confirmed: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: false,
    },
    contestId: {
      type: DataTypes.UUID,
      unique: uniqueIndex,
    },
    prize: {
      type: DataTypes.INTEGER,
    },
    rank: {
      type: DataTypes.INTEGER,
    },
  };
```

right?

Reply:
yes